pkg/api/api.go

```diff
@@ -76,6 +164,8 @@ func CreateMux(apiFunctions ScratchDataAPI) *chi.Mux {
 	api.Post("/data/insert/{table}", apiFunctions.Insert)
 	api.Get("/data/query", apiFunctions.Select)
 	api.Post("/data/query", apiFunctions.Select)
+	api.Post("/data/query/share", apiFunctions.CreateQuery)        // New endpoint for creating a query
+	api.Get("/share/{uuid}/data.{format}", apiFunctions.ShareData) // New endpoint for sharing data
```

Comment:
This path should be `/share/{uuid}...`. However, this route is `/api/share...` which is not what we want.

```suggestion
	r.Get("/share/{uuid}/data.{format}", apiFunctions.ShareData) // New endpoint for sharing data
```

Reply:
In order to make this work, perhaps the `r.Use(apiFunctions.AuthMiddleware)` should instead be `api.User(...)`.